src/ima.js

```diff
@@ -1098,8 +1098,15 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
    * @memberof Ima
    */
   _maybeForceExitFullScreen(): void {
-    const isIOS = () => this.player.env.os.name === 'iOS';
-    if (isIOS() && !this._adsManager.isCustomPlaybackUsed() && this.player.isFullscreen()) {
+    const isIOS = this.player.env.os.name === 'iOS';
+    //check if inBrowserFullscreen not set, just in case of inline true and not inBrowserFullscreen we will exit otherwise
```

Comment:
this sentence is not clear, the context of `we'll keep the original behavior` refers to the previous code before your change and not what the method is doing.
The sole purpose of this method is to do what you are referring to in the comment and it is already said in the JSDoc method description and it mentions `native full screen`
I think you can remove the comment.

Changing this will be more clear that you want to handle fullscreen but only when it's not inBrowserFullscreen:
```javascript
this.player.isFullscreen() &&
      (this.player.config.playback.playsinline && !this.player.config.playback.inBrowserFullscreen)
```

to this:
```javascript
(this.player.isFullscreen() && !this.player.config.playback.inBrowserFullscreen) &&
this.player.config.playback.playsinline
```

Reply:
That's right but it should be combination of fullscreen and not inBrowserFullscreen and plays inline, it's not enough to be fullscreen and not inBrowserFullscreen for this situation.
I'll change it

Reply:
it needs to be combination of all conditions in this long `if`.
What I mean is that each line in the long condition has some context.
It will be easier to understand the fullscreen context if you put in one line only the fullscreen conditionals.
```javascript
    if (
      isIOS &&
      !this._adsManager.isCustomPlaybackUsed() &&
      (this.player.isFullscreen() && !this.player.config.playback.inBrowserFullscreen) &&
      this.player.config.playback.playsinline
    ) {
      this.player.exitFullscreen();
    }
```